### ISSUE TITLE
Use of BindFunction

### ISSUE THREAD

**Issue by fredericoregateiro** [praeclarum/sqlite-net#574]
Hello,
Is there a way to bind a custom function? 

Like on System.Data.Sqlite we can use BindFunction on a SQLiteConnection

Thanks
Frederico

**Comment by praeclarum** [comment]
I've been meaning to add support for this but it just hasn't happened. I'll look into it!

**Comment by praeclarum** [comment]
Related to #547

**Comment by MKuckert** [comment]
I was able to add a first draft of an implementation when the `SQLitePCL.raw` library is used: https://github.com/MKuckert/sqlite-net/commit/21adf177d5426a227d81b94cf912b1776a22248d
But this only works if `USE_SQLITEPCL_RAW` is set – the other bindings provide no pre-bound `sqlite3_create_function` function.

Are the other configurations (`USE_CSHARP_SQLITE`, `USE_WP8_NATIVE_SQLITE`, none) in active use or just unused legacy code? I fear the former is the case…?

**Comment by praeclarum** [comment]
Yeah `USE_CSHARP_SQLITE` and `USE_WP8_NATIVE_SQLITE` are legacies from when I distributed this library as source code.

Now that I'm heavily pushing the Raw version, they make less sense. Might be time to yank them.

However, there is the last scenario of using the library with direct bindings to sqlite (mostly used on iOS and in my apps ;-)). It would be nice to support function binding there too.

**Comment by praeclarum** [comment]
@MKuckert I like your code!

The biggest trick is to support iOS AOT: https://developer.xamarin.com/guides/ios/advanced_topics/limitations/#Reverse_Callbacks

AFAIK your code should work - just need to test it on an actual device. (The SQLite.Tests.iOS project.)

**Comment by MKuckert** [comment]
@ericsink does the required trickery with the delegate callback to get that to work on iOS as far as I can judge.
I'll try to copy the required code from Eric in case of native iOS support.
Are you going to remove the legacy flags or create an up-for-grabs issue?

**Comment by praeclarum** [comment]
I made an up for grabs issue #609 since I'm plugging away on some bugs now.

**Comment by ericsink** [comment]
Yes, SQLitePCL.raw *should* do the right thing for custom functions, even on iOS. Do feel free to let me know if you have any trouble with this feature.

**Comment by x37v** [comment]
curious what the state of this is, been a while now, does it make sense to merge in?

**Comment by Kasta11111** [comment]
Any progress guys? We really need this.
